kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/assertions/timing/ContinuallyTest.kt

```diff
@@ -1,51 +1,63 @@
+@file:Suppress("DEPRECATION")
+
 package com.sksamuel.kotest.assertions.timing
 
 import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.assertions.throwables.shouldThrowAny
 import io.kotest.assertions.throwables.shouldThrowExactly
 import io.kotest.assertions.timing.continually
+import io.kotest.common.nonConstantFalse
+import io.kotest.common.nonConstantTrue
 import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.comparables.shouldBeLessThan
 import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldMatch
+import kotlinx.coroutines.delay
 import kotlin.time.Duration.Companion.hours
 import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.Duration.Companion.seconds
 
 class ContinuallyTest : WordSpec() {
 
-  init {
-    "continually" should {
-      "pass working tests" {
-          continually(500.milliseconds) {
-              (System.currentTimeMillis() > 0) shouldBe true
-          }
-      }
-      "fail broken tests immediately"  {
-        shouldThrowAny {
-          continually(12.hours) {
-            false shouldBe true
-          }
-        }
-      }
-      "fail should throw the underlying error" {
-        shouldThrowExactly<AssertionError> {
-          continually(12.hours) {
-            throw AssertionError("boom")
-          }
-        }.message shouldBe "boom"
-      }
-      "fail tests start off as passing then fail within the period" {
-        var n = 0
-        val e = shouldThrow<Throwable> {
+   init {
+      coroutineTestScope = true
+
+      "continually" should {
+         "pass working tests" {
+            continually(500.milliseconds) {
+               nonConstantTrue() shouldBe true
+            }
```

Comment:
Just to double check, is this PR supposed to make the deprecated `continually` use virtual time? I'm asking because test takes 500ms, but I'm not sure about whether it's supposed to use virtual time or not.

Reply:
Good catch! In fact, continually is still using `TimeSource.Monotonic`, so the above is wrong. I'll try to change continually to `testTimeSource()` in order to avoid bothering us with flaky tests while continually is still around.